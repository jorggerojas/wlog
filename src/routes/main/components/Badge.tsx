// @flow
import React from "react";
import { Link, BrowserRouter as Router } from "react-router-dom";

interface BadgeProps {
  title: string;
  isActive: boolean;
  link?: boolean;
  target?: string;
}
const Badge = ({ title, isActive, link, target }: BadgeProps) => {
  return (
    <li key={title} className={isActive ? "uk-active" : ""}>
      <Router>
        {link ? (
          <Link to="" uk-switcher-item={target ? `${target}` : ""}>
            {title}
          </Link>
        ) : (
          <Link to={""}>{title}</Link>
        )}
      </Router>
    </li>
  );
};

export default Badge;
