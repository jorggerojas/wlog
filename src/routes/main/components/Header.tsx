import React, { useState } from "react";
import { /*globalDark,*/ globalLight } from "../../../styles/global";
import { Link } from "react-router-dom";
import cookie from "react-cookies";

const Header = () => {
  const [user] = useState(cookie.load("USER", false) ?? "0");
  const { navColor, padd, whiteText } = globalLight();
  return (
    <div uk-sticky="sel-target: .uk-navbar-container; cls-active: uk-navbar-sticky">
      <nav
        className={`${navColor} uk-navbar uk-navbar-container uk-margin`}
        uk-navbar="mode:click"
      >
        <div className="uk-navbar-left">
          <Link className={`${whiteText} uk-navbar-item uk-logo`} to="/">
            Wlog
          </Link>
        </div>
        <div className="uk-navbar-right">
          <ul className="uk-navbar-nav">
            <li className={padd}>
              {user === "0" ? (
                <Link to="/sign" className={whiteText}>
                  Iniciar sesi√≥n
                </Link>
              ) : (
                <Link to={`user/${user}`} className={whiteText}>
                  {user.toUpperCase()}
                </Link>
              )}
            </li>
          </ul>
        </div>
        <div className="uk-navbar-center">
          <Link className={`uk-navbar-toggle ${whiteText}`} to="#">
            <span uk-icon="icon:plus-circle"></span>
          </Link>
        </div>
      </nav>
    </div>
  );
};

export default Header;
