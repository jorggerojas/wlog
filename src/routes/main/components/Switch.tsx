import React from "react";

interface SwitchProps {
  check: boolean;
  checked: Function;
}

const Switch = ({ check, checked }: SwitchProps) => {
  const change = () => {
    checked(!check);
  };
  return (
    <div className="mid">
      <label className="rocker rocker-small">
        <input type="checkbox" checked={check} onChange={change} />
        <span className="switch-left">🌚</span>
        <span className="switch-right">🌞</span>
      </label>
    </div>
  );
};

export default Switch;
