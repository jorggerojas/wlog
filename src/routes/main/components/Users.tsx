import axios from "axios";
import React, { useEffect, useState } from "react";
import DataTable from "react-data-table-component";
import { loadStorage, URL } from "../../../config";
import { options, customStyles } from "./tableOptions";
import { Select, Option } from "../../../styles/text";

interface UserProps {
  theme?: boolean;
}

const Users = ({ theme }: UserProps) => {
  const [disabled, setDisabled] = useState(false);
  const [data, setData] = useState([]);
  const [totalElements, setTotalElements] = useState(0);
  const getUsers = (page?: number) => {
    return axios.get(`${URL}/users?page=${page ?? 0}&size=10`);
  };
  const changeValue = (value: string, username: string) => {
    setDisabled(true);
    setTimeout(() => {
      axios({
        method: "put",
        url: `${URL}/users/${username}`,
        headers: {
          Authorization: `Bearer ${loadStorage("TOKEN")}`,
        },
        data: {
          role: [`${value}`],
        },
      }).then(() => {
        getUsers()
          .then(({ data }: any) => {
            setData(
              data.content.map((user: any) => {
                return {
                  key: user.nickname,
                  role: user.role[0],
                  name: user.name,
                  username: user.nickname,
                  datelog: user.dateLog,
                };
              })
            );
            setTotalElements(data.totalElements);
            setDisabled(false);
          })
          .catch(() => {
            setData([]);
          });
      });
    }, 500);
  };
  const columns = [
    {
      name: "Role",
      selector: "role",
      sortable: true,
      cell: (row: any) => {
        return (
          <Select
            defaultValue={row.role ?? "REDACTOR"}
            className={" uk-text-bold"}
            onChange={(e) => changeValue(e.target.value, row.username ?? "...")}
          >
            <Option value="ADMIN">ADMIN</Option>
            <Option value="REDACTOR">REDACTOR</Option>
            <Option value="MODERADOR">MODERADOR</Option>
            <Option value="LECTOR">LECTOR</Option>
          </Select>
        );
      },
    },
    {
      name: "Full name",
      selector: "name",
      hide: 599,
      sortable: true,
      cell: (row: any) => {
        return row ? (
          <div className="uk-flex uk-flex-middle uk-flex-center">
            <p className="uk-text-bold">{row.name}</p>
          </div>
        ) : null;
      },
    },
    {
      name: "Username",
      selector: "username",
      sortable: true,
      cell: (row: any) => {
        return row ? (
          <p className="uk-text-bold">
            {row && row.username ? row.username.toUpperCase() : "..."}
          </p>
        ) : null;
      },
    },
    {
      name: "Date log",
      selector: "datelog",
      sortable: true,
      hide: 959,
      cell: (row: any) => {
        return row ? (
          <p className="uk-text-bold">{row ? row.datelog : "..."}</p>
        ) : null;
      },
    },
  ];
  useEffect(() => {
    getUsers()
      .then(({ data }: any) => {
        setData(
          data.content.map((user: any) => {
            return {
              key: user.nickname,
              role: user.role[0],
              name: user.name,
              username: user.nickname,
              datelog: user.dateLog,
            };
          })
        );
        setTotalElements(data.totalElements);
      })
      .catch(() => {
        setData([]);
      });
  }, []);
  const changeContent = (page: any) => {
    getUsers(page)
      .then(({ data }: any) => {
        setData(
          data.content.map((user: any) => {
            return {
              key: user.nickname,
              role: user.role[0],
              name: user.name,
              username: user.nickname,
              datelog: user.dateLog,
            };
          })
        );
        setTotalElements(data.totalElements);
      })
      .catch(() => {
        setData([]);
      });
  };
  return (
    <div className="uk-padding-small ">
      <DataTable
        data={data}
        columns={columns}
        responsive
        dense={false}
        disabled={disabled}
        pagination
        paginationServer
        paginationTotalRows={totalElements}
        paginationComponentOptions={options}
        customStyles={customStyles(theme)}
        className="uk-table"
        onChangePage={changeContent}
      />
    </div>
  );
};

export default Users;
