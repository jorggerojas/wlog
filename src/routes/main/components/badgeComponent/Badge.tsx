import React /*useState*/ from "react";

interface BadgeProps {
  data: Array<Object>;
}
const Badge = ({ data }: BadgeProps) => {
  //   const [active, setActive] = useState("false");
  return (
    <ul className="uk-subnav uk-subnav-pill uk-margin-small-left">
      {data.length
        ? data.map((badge: Object) => {
            const pill = JSON.parse(JSON.stringify(badge));
            return (
              <li key={pill.title} className={pill.isActive ? "uk-active" : ""}>
                <a href="#s">{pill.title}</a>
              </li>
            );
          })
        : null}
    </ul>
  );
};

export default Badge;
