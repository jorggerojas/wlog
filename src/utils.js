import React from "react";
import { render, waitFor } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import "@testing-library/jest-dom";

const renderWithRouter = (ui, { route = "/", ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  const finishLoading = () =>
    waitFor(() =>
      expect(utils.queryByText("Loading, please wait...")).toBeNull()
    );
  return {
    ...utils,
    finishLoading,
    history,
  };
};

export {
  Simulate,
  wait,
  render,
  cleanup,
  renderIntoDocument,
  fireEvent,
} from "@testing-library/react";
export { renderWithRouter };

export function renderWithProviders(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
  apolloMocks
) {
  console.log("I am route inside renderWithProviders Wrapper:", route);
  return {
    ...render(
      <Router history={history}>
        <Route path={route}>{ui}</Route>
      </Router>
    ),
    history,
  };
}
