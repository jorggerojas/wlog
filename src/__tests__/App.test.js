import React from "react";
import { cleanup } from "@testing-library/react";
import axiosMock from "axios";
import Badge from "../routes/main/components/Badge";
import Header from "../routes/main/components/Header";
import Switch from "../routes/main/components/Switch";
import Users from "../routes/main/components/Users";
import { renderWithRouter } from "../utils";

afterEach(cleanup);
jest.mock("axios");

describe("<App/> tree", () => {
  describe("<Header/> component", () => {
    test("render component without session", async () => {
      const { container } = renderWithRouter(<Header />);
      await expect(container.innerHTML).toMatch("Log in");
      await expect(container).toMatchSnapshot();
    });
    test("render component with session", async () => {
      const { container } = renderWithRouter(<Header username={"FakeUser1"} />);
      await expect(container.innerHTML).not.toMatch("Log in");
      await expect(container.innerHTML).toMatch("FakeUser1");
      await expect(container).toMatchSnapshot();
    });
  });
  describe("<Badge/> component", () => {
    test("render component", async () => {
      const props = {
        title: "USERS",
        isActive: true,
        link: true,
      };
      const { container } = renderWithRouter(<Badge {...props} />);
      await expect(container.firstChild).toHaveClass("uk-active");
      await expect(container.innerHTML).toMatch(props.title);
      await expect(container).toMatchSnapshot();
    });
  });
  describe("<Users/> component", () => {
    test("render component", async () => {
      const fakeUsers = [
        {
          key: "JorgeRojas",
          role: "ADMIN",
          name: "Jorge Rojas",
          username: "JorgeRojas",
          datelog: "2020-12-12",
        },
        {
          key: "mamberroi",
          role: "LECTOR",
          name: "Mamberroi One",
          username: "mamberroi",
          datelog: "2019-10-12",
        },
      ];
      axiosMock.get.mockImplementationOnce(() =>
        Promise.resolve({
          data: { content: fakeUsers },
        })
      );
      const { container } = renderWithRouter(<Users />);
      await expect(axiosMock.get).toHaveBeenCalledTimes(1);
      await expect(container.innerHTML).toMatch("LECTOR");
      await expect(container.innerHTML).toMatch("ADMIN");
      await expect(container.innerHTML).toMatch("Role");
      await expect(container).toMatchSnapshot();
    });
  });
  describe("<Switch/> component", () => {
    test("render component and change the input value", async () => {
      const { container } = renderWithRouter(
        <Switch check={false} checked={() => {}} />
      );
      await expect(container.firstChild.firstChild.firstChild.checked).toBe(
        false
      );
      const check = (container.firstChild.firstChild.firstChild.checked = true);
      await expect(check).toBe(true);
    });
  });
});
