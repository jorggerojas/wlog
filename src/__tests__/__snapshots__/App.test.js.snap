// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> tree <Badge/> component render component 1`] = `
<div>
  <li
    class="uk-active"
  >
    <a
      href="/"
      uk-switcher-item=""
    >
      USERS
    </a>
  </li>
</div>
`;

exports[`<App/> tree <Header/> component render component with session 1`] = `
<div>
  <div
    uk-sticky="sel-target: .uk-navbar-container; cls-active: uk-navbar-sticky"
  >
    <nav
      class="sc-dlfnbm gvpgwc uk-navbar uk-navbar-container uk-margin"
    >
      <div
        class="uk-navbar-left"
      >
        <a
          class="uk-navbar-item uk-logo"
          href="/"
        >
          <span
            class="sc-fodVxV frAqKJ"
          >
            Wlog
          </span>
        </a>
      </div>
      <div
        class="uk-navbar-right"
      >
        <ul
          class="uk-navbar-nav"
        >
          <li
            class="uk-visible"
          >
            <div
              class="mid"
            >
              <label
                class="rocker rocker-small"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="switch-left"
                >
                  ðŸŒš
                </span>
                <span
                  class="switch-right"
                >
                  ðŸŒž
                </span>
              </label>
            </div>
          </li>
          <li
            class="uk-visible@s"
          >
            <a
              href="/user/RealUser1"
            >
              <span
                class="sc-fodVxV frAqKJ"
              >
                REALUSER1
              </span>
            </a>
          </li>
          <li
            class="uk-hidden@s"
          >
            <a
              href="/user/RealUser1"
            >
              <span
                class="sc-fodVxV frAqKJ"
              >
                REALUSER1
              </span>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="uk-navbar-center"
      >
        <a
          class="uk-navbar-toggle"
          href="/create"
        >
          <span
            class="white"
            uk-icon="icon:plus-circle"
          />
        </a>
      </div>
    </nav>
  </div>
</div>
`;

exports[`<App/> tree <Header/> component render component without session 1`] = `
<div>
  <div
    uk-sticky="sel-target: .uk-navbar-container; cls-active: uk-navbar-sticky"
  >
    <nav
      class="sc-dlfnbm gvpgwc uk-navbar uk-navbar-container uk-margin"
    >
      <div
        class="uk-navbar-left"
      >
        <a
          class="uk-navbar-item uk-logo"
          href="/"
        >
          <span
            class="sc-fodVxV frAqKJ"
          >
            Wlog
          </span>
        </a>
      </div>
      <div
        class="uk-navbar-right"
      >
        <ul
          class="uk-navbar-nav"
        >
          <li
            class="uk-visible"
          >
            <div
              class="mid"
            >
              <label
                class="rocker rocker-small"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="switch-left"
                >
                  ðŸŒš
                </span>
                <span
                  class="switch-right"
                >
                  ðŸŒž
                </span>
              </label>
            </div>
          </li>
          <li
            class="uk-visible@s"
          >
            <a
              class=""
              href="/sign"
            >
              <span
                class="sc-fodVxV frAqKJ"
              >
                Log in
              </span>
            </a>
          </li>
          <li
            class="uk-hidden@s"
          >
            <span
              class="sc-fodVxV frAqKJ"
            >
               
              <a
                href="/sign"
                uk-icon="icon:user; ratio:1.3"
              />
            </span>
          </li>
        </ul>
      </div>
      <div
        class="uk-navbar-center"
      />
    </nav>
  </div>
</div>
`;

exports[`<App/> tree <Users/> component render component 1`] = `
<div>
  <div
    class="uk-padding-small "
  >
    <div
      aria-level="1"
      class="sc-eJMQSu kgKbGY rdt_TableHeader"
      role="heading"
    >
      <div
        class="sc-nFpLZ flEpPN"
      />
      <div
        class="sc-higXBA jiKwoX"
      />
      <div
        class="sc-jeGSBP dVvwlT"
      >
        <div
          class="sc-laRPJI iAlzeP"
        />
        <div
          class="sc-iNqMTl gmpGVg"
        />
      </div>
    </div>
    <div
      class="sc-gGmIRh jcZxkp uk-table"
    >
      <div
        class="sc-fbkhIv hKqcFy"
      >
        <div
          class="sc-jcVebW kFNPkp rdt_Table"
          role="table"
        >
          <div
            class="sc-bZSQDF ZrjgT rdt_TableHead"
            role="rowgroup"
          >
            <div
              class="sc-iUuytg cBMAXA rdt_TableHeadRow"
              role="row"
            >
              <div
                class="sc-iBaPrD sc-eggNIi sc-hTZhsR cWAnjz erMqUn kAJLoH rdt_TableCol"
              >
                <div
                  class="sc-jgHCyG kiZQaB rdt_TableCol_Sortable"
                  id="column-role"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    Role
                  </div>
                  <span
                    class="sc-khAkjo dnpBrq"
                  >
                    â–²
                  </span>
                </div>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-hTZhsR cWAnjz hRbKtS kAJLoH rdt_TableCol"
              >
                <div
                  class="sc-jgHCyG kiZQaB rdt_TableCol_Sortable"
                  id="column-name"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    Full name
                  </div>
                  <span
                    class="sc-khAkjo dnpBrq"
                  >
                    â–²
                  </span>
                </div>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-hTZhsR cWAnjz erMqUn kAJLoH rdt_TableCol"
              >
                <div
                  class="sc-jgHCyG kiZQaB rdt_TableCol_Sortable"
                  id="column-username"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    Username
                  </div>
                  <span
                    class="sc-khAkjo dnpBrq"
                  >
                    â–²
                  </span>
                </div>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-hTZhsR cWAnjz eBYGQa kAJLoH rdt_TableCol"
              >
                <div
                  class="sc-jgHCyG kiZQaB rdt_TableCol_Sortable"
                  id="column-datelog"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    Date log
                  </div>
                  <span
                    class="sc-khAkjo dnpBrq"
                  >
                    â–²
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-lcujXC hkBqdn rdt_TableBody"
            offset="250px"
            role="rowgroup"
          >
            <div
              class="sc-AzgDb kKqGYG rdt_TableRow"
              id="row-0"
              role="row"
            >
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ erMqUn brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-wKK1OPHP5w-undefined"
                role="gridcell"
              >
                <select
                  class="sc-crrsfI gnBEvf  uk-text-bold"
                >
                  <option
                    class="sc-dQppl eDVlRn"
                    value="ADMIN"
                  >
                    ADMIN
                  </option>
                  <option
                    class="sc-dQppl eDVlRn"
                    value="REDACTOR"
                  >
                    REDACTOR
                  </option>
                  <option
                    class="sc-dQppl eDVlRn"
                    value="MODERADOR"
                  >
                    MODERADOR
                  </option>
                  <option
                    class="sc-dQppl eDVlRn"
                    value="LECTOR"
                  >
                    LECTOR
                  </option>
                </select>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ hRbKtS brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-lFnVVCNat6-undefined"
                role="gridcell"
              >
                <div
                  class="uk-flex uk-flex-middle uk-flex-center"
                >
                  <p
                    class="uk-text-bold"
                  >
                    Jorge Rojas
                  </p>
                </div>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ erMqUn brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-uX1IoK4nmc-undefined"
                role="gridcell"
              >
                <p
                  class="uk-text-bold"
                >
                  ...
                </p>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ eBYGQa brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-AdvzllAkYV-undefined"
                role="gridcell"
              >
                <p
                  class="uk-text-bold"
                />
              </div>
            </div>
            <div
              class="sc-AzgDb kKqGYG rdt_TableRow"
              id="row-1"
              role="row"
            >
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ erMqUn brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-wKK1OPHP5w-undefined"
                role="gridcell"
              >
                <select
                  class="sc-crrsfI gnBEvf  uk-text-bold"
                >
                  <option
                    class="sc-dQppl eDVlRn"
                    value="ADMIN"
                  >
                    ADMIN
                  </option>
                  <option
                    class="sc-dQppl eDVlRn"
                    value="REDACTOR"
                  >
                    REDACTOR
                  </option>
                  <option
                    class="sc-dQppl eDVlRn"
                    value="MODERADOR"
                  >
                    MODERADOR
                  </option>
                  <option
                    class="sc-dQppl eDVlRn"
                    value="LECTOR"
                  >
                    LECTOR
                  </option>
                </select>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ hRbKtS brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-lFnVVCNat6-undefined"
                role="gridcell"
              >
                <div
                  class="uk-flex uk-flex-middle uk-flex-center"
                >
                  <p
                    class="uk-text-bold"
                  >
                    Mamberroi One
                  </p>
                </div>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ erMqUn brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-uX1IoK4nmc-undefined"
                role="gridcell"
              >
                <p
                  class="uk-text-bold"
                >
                  ...
                </p>
              </div>
              <div
                class="sc-iBaPrD sc-eggNIi sc-cTkwdZ eWtMhQ eBYGQa brFPfv rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-AdvzllAkYV-undefined"
                role="gridcell"
              >
                <p
                  class="uk-text-bold"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <nav
        class="sc-jHVexB gFgSGo rdt_Pagination"
      >
        <span
          class="sc-hlTvYk sc-bTvRPi kPbfwi fogUgU"
        >
          Filas por pÃ¡gina:
        </span>
        <div
          class="sc-TmcTc edKdoe"
        >
          <select
            aria-label="Filas por pÃ¡gina:"
            class="sc-dwfUOf eNIwRb"
          >
            <option
              selected=""
              value="10"
            >
              10
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="25"
            >
              25
            </option>
            <option
              value="30"
            >
              30
            </option>
          </select>
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
        <span
          class="sc-hlTvYk sc-eLgOdN kPbfwi kHZoSK"
        >
          1-2 de 2
        </span>
        <div
          class="sc-jONnTn jIdfvt"
        >
          <button
            aria-disabled="true"
            aria-label="First Page"
            class="sc-aemoO guPiQE"
            disabled=""
            id="pagination-first-page"
            type="button"
          >
            <svg
              aria-hidden="true"
              height="24"
              role="presentation"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
              />
              <path
                d="M24 24H0V0h24v24z"
                fill="none"
              />
            </svg>
          </button>
          <button
            aria-disabled="true"
            aria-label="Previous Page"
            class="sc-aemoO guPiQE"
            disabled=""
            id="pagination-previous-page"
            type="button"
          >
            <svg
              aria-hidden="true"
              height="24"
              role="presentation"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
          <button
            aria-disabled="true"
            aria-label="Next Page"
            class="sc-aemoO guPiQE"
            disabled=""
            id="pagination-next-page"
            type="button"
          >
            <svg
              aria-hidden="true"
              height="24"
              role="presentation"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
          <button
            aria-disabled="true"
            aria-label="Last Page"
            class="sc-aemoO guPiQE"
            disabled=""
            id="pagination-last-page"
            type="button"
          >
            <svg
              aria-hidden="true"
              height="24"
              role="presentation"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </button>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
